src/fsharp/service/service.fs

```diff
@@ -2656,20 +2656,26 @@ type BackgroundCompiler(legacyReferenceResolver, projectCacheSize, keepAssemblyC
         let execWithReactorAsync action = reactor.EnqueueAndAwaitOpAsync(userOpName, "ParseAndCheckFileInProject", filename, action)
         async {
             try 
+                Logger.BlockStart(LogCompilerFunctionId.Parse_And_Check_File_In_Project)
```

Comment:
Consider making an `IDisposable` and using `using` statements.  Otherwise an exception will mean you won't get matching `BlockStop` calls.

Reply:
Makes sense. I just checked and Roslyn does the same thing.